cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/dataset/SearchRequest.java

```diff
@@ -0,0 +1,196 @@
+/*
+ * Copyright © 2018 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ */
+
+package co.cask.cdap.data2.metadata.dataset;
+
+import co.cask.cdap.proto.EntityScope;
+import co.cask.cdap.proto.element.EntityTypeSimpleName;
+import co.cask.cdap.proto.id.NamespaceId;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Objects;
+import java.util.Set;
+import javax.annotation.Nullable;
+
+/**
+ * A Metadata search request.
+ */
+public class SearchRequest {
+  private final NamespaceId namespace;
+  private final String query;
+  private final Set<EntityTypeSimpleName> types;
+  private final SortInfo sortInfo;
+  private final int offset;
+  private final int limit;
+  private final int numCursors;
+  @Nullable
+  private final String cursor;
+  private final boolean showHidden;
+  private final Set<EntityScope> entityScope;
+
+  public SearchRequest(NamespaceId namespace, String query, Set<EntityTypeSimpleName> types, SortInfo sortInfo,
+                       int offset, int limit, int numCursors, @Nullable String cursor, boolean showHidden,
+                       Set<EntityScope> entityScope) {
+    if (offset < 0) {
+      throw new IllegalArgumentException("offset must not be negative");
+    }
+    if (limit < 0) {
+      throw new IllegalArgumentException("limit must not be negative");
+    }
+    this.namespace = namespace;
+    this.query = query;
+    this.types = Collections.unmodifiableSet(new HashSet<>(types));
+    this.sortInfo = sortInfo;
+    this.numCursors = numCursors;
+    this.cursor = cursor;
+    this.showHidden = showHidden;
+    this.entityScope = Collections.unmodifiableSet(new HashSet<>(entityScope));
+    this.offset = offset;
+    this.limit = limit;
+  }
+
+  /**
+   * @return the namespace to search in
+   */
+  public NamespaceId getNamespace() {
```

Comment:
getNamespaceId() to make getNamespace() let different.

Reply:
done